### ISSUE TITLE
[Go] Convert for Kubernetes provider generates invalid code

### ISSUE THREAD

**Issue by lblackstone** [pulumi/pulumi-yaml#283]
### What happened?

`pulumi convert` generates invalid Go code for Kubernetes providers.

### Steps to reproduce

```yaml
name: pulumi-yaml-patch
runtime: yaml
description: A minimal Kubernetes Pulumi YAML program
resources:
  provider:
    type: pulumi:providers:kubernetes
    properties:
      enableServerSideApply: true
```

### Expected Behavior

The Kubernetes provider lives in the top-level package rather than in `/providers`.

```go
package main

import (
	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		provider, err := kubernetes.NewProvider(ctx, "provider", &kubernetes.ProviderArgs{
			EnableServerSideApply: true,
		})
		if err != nil {
			return err
		}
		return nil
	})
}
```

### Actual Behavior

```go
package main

import (
	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/providers"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		provider, err := providers.Newkubernetes(ctx, "provider", &providers.kubernetesArgs{
			EnableServerSideApply: true,
		})
		if err != nil {
			return err
		}
		return nil
	})
}
```

### Versions used

```
pulumi about
CLI
Version      3.35.3
Go Version   go1.18.3
Go Compiler  gc

Plugins
NAME        VERSION
kubernetes  unknown
yaml        unknown

Host
OS       darwin
Version  12.4
Arch     arm64

This project is written in yaml
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AaronFriel** [comment]
Related PR solving additional codegen bugs: https://github.com/pulumi/pulumi/pull/10150

**Comment by AaronFriel** [comment]
Kubernetes RFC for reference: https://github.com/pulumi/pulumi-kubernetes/issues/2011

**Comment by t0yv0** [comment]
Might have accidentally closed this based on pulumi-java#771 going in, but the scope might be across languages.

**Comment by aq17** [comment]
> Might have accidentally closed this based on pulumi-java#771 going in, but the scope might be across languages.

I think Java was the remaining language, maybe @lblackstone could confirm?

**Comment by lblackstone** [comment]
Yes, my understanding is that all languages should be fixed now.
